src/multiconfigwidget.h

```diff
@@ -11,32 +10,43 @@
 #include <KConfigGroup>
 #include <QWidget>
 
-class QComboBox;
-class QPushButton;
+namespace Ui {
+class MultiConfigWidget;
+}
 
+/** this widget allows fast switching between different configurations
+ * of m_child
+ * saveConfig will automatically save all changes to m_group
+ * use setChildWidget for an unparented one and moveWidgetToChild for a parented one
+ * */
 class MultiConfigWidget : public QWidget
 {
     Q_OBJECT
 public:
     MultiConfigWidget(QWidget* parent = nullptr);
-    ~MultiConfigWidget();
+    ~MultiConfigWidget() override;
 
-    QString currentConfig() const;
+    void setChildWidget(QWidget* widget, const QVector<QWidget*>& formWidgets);
```

Comment:
add basic API documentation to explain that `widget` is basically the content widget, and `formWidgets` are the individual widgets (that must have a unique object name!) which will get saved/loaded automatically